packages/user/TokenStream/service/src/tests.rs

```diff
@@ -90,6 +90,14 @@ mod tests {
         Ok(token_id)
     }
 
+    pub fn full_vesting_time(half_life_seconds: u32, balance: u64) -> i64 {
+        if balance == 0 {
+            return 0;
+        }
+        let t = half_life_seconds as f64 * ((2.0 * balance as f64).ln() / std::f64::consts::LN_2);
```

Comment:
Why is multiplying by 2.0 needed? The unvested balance is rounded down, so it should flush to zero when it drops below 1, not when it drops below 0.5. Also, there's `log2`.